nemo/collections/asr/models/sortformer_diar_models.py

```diff
@@ -269,6 +273,92 @@ def forward_infer(self, emb_seq):
         preds = self.sortformer_modules.forward_speaker_sigmoids(trans_emb_seq)
         return preds
 
+    def _diarize_forward(self, batch: Any):
+        """
+        A counterpart of `_transcribe_forward` function in ASR.
+        This function is a wrapper for forward pass functions for compataibility
+        with the existing classes.
+
+        Args:
+            batch (Any): The input batch containing audio signal and audio signal length.
+
+        Returns:
+            preds (torch.Tensor): Sorted tensor containing Sigmoid values for predicted speaker labels.
+                Shape: (batch_size, diar_frame_count, num_speakers)
+        """
+        with torch.no_grad():
+            preds = self.forward(audio_signal=batch[0], audio_signal_length=batch[1])
+            preds = preds.to('cpu')
+            torch.cuda.empty_cache()
+        return preds
+
+    def _diarize_output_processing(
+        self, outputs, uniq_ids, diarcfg: DiarizeConfig
+    ) -> Union[List[List[str]], Tuple[List[List[str]], List[torch.Tensor]]]:
+        preds_list, diar_output_lines_list = [], []
+        if outputs.shape[0] == 1:  # batch size = 1
+            preds_list.append(outputs)
+        else:
+            preds_list.extend(torch.split(outputs, [1] * outputs.shape[0]))
+
```

Comment:
Added docstrings.

We are aware that there is overlap between `predlist_to_timestamps` but we weren't able to merge this because of the Mixins class and unit-test. I added this into TODO memo so that we can handle this later on.